lib/saluki-app/src/logging.rs

```diff
@@ -61,7 +77,9 @@ pub fn fatal_and_exit(message: String) {
 /// # Errors
 ///
 /// If the logging subsystem was already initialized, an error will be returned.
-pub fn initialize_logging(default_level: Option<LevelFilter>) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+pub fn initialize_logging(
+    default_level: Option<LevelFilter>,
+) -> Result<WorkerGuard, Box<dyn std::error::Error + Send + Sync>> {
```

Comment:
I think there's enough things to be tweaked now (some of which I bring up in comments further down in the file) that we should probably expose a configuration object for all of the logging settings: default log level, whether or not to enable JSON logging, log file stuff, etc.

... and to complete the thought, take such a configuration as the one and only argument to this function.